juju/python-libjuju
# BundleHandler uses GetChangesMapArgs when available

The following PR updates the BundleHandler code to use `GetChangesMapArgs` from Juju when possible. Additionally also updated the schema to have the latest 2.7-dev schema.

### QA

Ensure you have the following branch checked out https://github.com/juju/juju/pull/10601, or if it's merged, use develop branch.

Deploy the branch so we can test against it:
```sh
juju bootstrap lxd test --no-gui --build-agent
```

To ensure that we're using the right method call, change the following:
```diff
diff --git a/juju/bundle.py b/juju/bundle.py
index f6fd91f..6df6635 100644
--- a/juju/bundle.py
+++ b/juju/bundle.py
@@ -140,7 +140,7 @@ class BundleHandler:
                 bundleurl=entity_id,
                 yaml=yaml.dump(self.bundle))
         else:
-            log.info("bundle_facade using GetChangesMapArgs API to deploy bundle changes")
+            log.critical("bundle_facade using GetChangesMapArgs API to deploy bundle changes")
             self.plan = await self.bundle_facade.GetChangesMapArgs(
                 bundleurl=entity_id,
                 yaml=yaml.dump(self.bundle))
```

Then run:
```sh
tox -e example -- examples/deploy_bundle.py
```

You should see in the output of the example:
```sh
Connecting to model
Deploying bundle
bundle_facade using GetChangesMapArgs API to deploy bundle changes
Waiting for active
Successfully deployed!
Removing bundle
Disconnecting from model
Success
```
